store/src/main/java/com/dropbox/android/external/store4/StoreBuilder.kt

```diff
@@ -137,135 +144,93 @@ interface StoreBuilder<Key : Any, Output : Any> {
-        val sourceOfTruth: PersistentNonFlowingSourceOfTruth<Key, Output, Output> =
-            PersistentNonFlowingSourceOfTruth(
-                realReader = { key -> persister.read(key) },
-                realWriter = { key, input -> persister.write(key, input) },
-                realDelete = { error("Delete is not implemented in legacy persisters") },
-                realDeleteAll = { error("Delete all is not implemented in legacy persisters") }
-            )
-        return withLegacySourceOfTruth(sourceOfTruth)
-    }
+        /**
+         * Creates a new [StoreBuilder] from a [Flow] fetcher.
+         *
+         * Use when creating a [Store] that fetches objects in an websocket-like multiple responses
+         * per request protocol.
+         *
+         * @param fetcher a function for fetching a flow of network records.
+         * @param sourceOfTruth a [SourceOfTruth] for the store.
+         */
+        fun <Key : Any, Input : Any, Output : Any> from(
+            fetcher: (key: Key) -> Flow<Input>,
+            sourceOfTruth: SourceOfTruth<Key, Input, Output>
+        ): StoreBuilder<Key, Output> = RealStoreBuilder(
+            fetcher = fetcher,
+            fetcherTransformer = { FetcherResult.Data(it) },
+            sourceOfTruth = sourceOfTruth
+        )
 
-    override fun <NewOutput : Any> persister(
-        reader: (Key) -> Flow<NewOutput?>,
-        writer: suspend (Key, Output) -> Unit,
-        delete: (suspend (Key) -> Unit)?,
-        deleteAll: (suspend () -> Unit)?
-    ): BuilderWithSourceOfTruth<Key, Output, NewOutput> {
-        return withSourceOfTruth(
-            PersistentSourceOfTruth(
-                realReader = reader,
-                realWriter = writer,
-                realDelete = delete,
-                realDeleteAll = deleteAll
-            )
+        /**
+         * Creates a new [StoreBuilder] from a [Flow] fetcher.
+         *
+         * Use when creating a [Store] that fetches objects in an websocket-like multiple responses
+         * per request protocol.
+         *
+         * @param fetcher a function for fetching a flow of network records.
+         * @param fetcherTransformer used to translate your fetcher's return value to success value
+         * or error in the case that your fetcher does not communicate errors through exceptions
+         */
+        fun <Key : Any, RawOutput : Any, Output : Any> from(
+            fetcher: (key: Key) -> Flow<RawOutput>,
+            fetcherTransformer: (RawOutput) -> FetcherResult<Output>
+        ): StoreBuilder<Key, Output> = RealStoreBuilder(
+            fetcher = fetcher,
+            fetcherTransformer = fetcherTransformer
         )
-    }
 
-    override fun build(): Store<Key, Output> {
-        return withSourceOfTruth<Output>().build()
+        /**
+         * Creates a new [StoreBuilder] from a [Flow] fetcher.
+         *
+         * Use when creating a [Store] that fetches objects in an websocket-like multiple responses
```

Comment:
websocket seems weird example as people may not be able to relate to it (i might be wrong). Maybe we can use firebase as an example instead?